File: SwiftApp/SwiftApp/Lap.swift
//
//  Copyright (c) 2015 Harry Cheung
//

import UIKit

final class Lap: NSObject {
  
  let GATE_RANGE: Double = 100
  
  weak var session:  Session?
  weak var track:    Track?
  var duration: Double
  var distance: Double
  var valid:    Bool
  let startTime: Double
  let lapNumber: Int
  var points:   [Point]
  let outLap:   Bool
  var splits:   [Double]
  
  init(session: Session, track: Track, startTime: Double, lapNumber: Int) {
    self.session   = session
    self.track     = track
    self.startTime = startTime
    self.lapNumber = lapNumber
    self.points    = []
    self.duration  = 0
    self.distance  = 0
    self.valid     = false
    self.outLap    = lapNumber == 0
    self.splits    = [Double](count: track.numSplits(), repeatedValue: 0)
  }
  
  func add(point: Point) {
    duration = point.lapTime
    distance = point.lapDistance
    points.append(point)
  }
  
}

File: JavaScript/lap.js
'use strict';

//
// Copyright (c) 2015 Harry Cheung
//

function Lap(session, track, startTime, lapNumber) {
  this.session = session;
  this.track = track;
  this.startTime = startTime;
  this.lapNumber = lapNumber;
  this.points = [];
  this.duration = 0;
  this.distance = 0;
  this.valid = false;
  this.outLap = lapNumber === 0;
  this.splits = [];
  var size = track.numSplits();
  while (size--) {
    this.splits.push(0);
  }
}

Lap.prototype.add = function (point) {
  this.duration = point.lapTime;
  this.distance = point.lapDistance;
  this.points.push(point);
};

if (typeof exports !== 'undefined' && typeof module !== 'undefined' && module.exports) {
  exports = module.exports = Lap;
}
